subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -180,6 +181,7 @@ void add_ipv6_addr_to_ot(struct openthread_context *context)
 			break;
 		}
 	}
+	i = i < 0 ? 0 : i;
```

Comment:
I don't think this is a good fix, if the loop hasn't found any address we cannon proceed with some default entry. I'd exit the function here in case `i` is negative. Anyway, I don't think this should ever happen unless there's some bug (we only call this function on  `NET_EVENT_IPV6_ADDR_ADD` event), so perhaps even an assert wouldn't be a bad idea.